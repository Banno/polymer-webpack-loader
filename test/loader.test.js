/* eslint no-undefined: "off", no-useless-escape: "off" */

import sourceMap from 'source-map';
import loader from '../src';

const normalisePaths = result => result.replace('src\\\\', 'src/');

function verifySourceMap(generatedSource, map) {
  const consumer = sourceMap.SourceMapConsumer(map);
  consumer.eachMapping((mapping) => {
    if (!mapping.name) {
      return;
    }
    const originalSourceByLine = consumer.sourceContentFor(mapping.source).split('\n');
    const generatedSourceByLine = generatedSource.split('\n');

    expect(generatedSourceByLine[mapping.generatedLine - 1].substr(mapping.generatedColumn, mapping.name.length))
      .toBe(mapping.name);
    expect(originalSourceByLine[mapping.originalLine - 1].substr(mapping.originalColumn, mapping.name.length))
      .toBe(mapping.name);
  });
}

describe('loader', () => {
  let opts;

  beforeEach(() => {
    opts = {
      resourcePath: 'src/test.html',
      query: {
        htmlLoader: {
          minimize: false,
          exportAsDefault: true,
          exportAsEs6Default: true,
        },
      },
    };
  });

  test('can process basic input', (done) => {
    opts.async = () => (err, source, map) => {
      expect(err).toBe(null);
      expect(normalisePaths(source)).toMatchSnapshot();
      expect(map).toBe(undefined);
      done();
    };
    loader.call(opts, '<div></div>');
  });

  test('can process without options', (done) => {
    opts.query = null;
    opts.async = () => (err, source, map) => {
      expect(err).toBe(null);
      expect(normalisePaths(source)).toMatchSnapshot();
      expect(map).toBe(undefined);
      done();
    };
    loader.call(opts, '<div></div>');
  });

  describe('links', () => {
    test('transforms links', (done) => {
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<link rel="import" href="foo.html">');
    });

    test('ignores links with invalid href', (done) => {
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<link rel="import" href="">');
    });

    test('ignores stylesheet links with external href', (done) => {
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<link rel="stylesheet" href="https://example.com/foo.html">');
    });

    test('ignores import links with external href', (done) => {
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<link rel="import" href="https://example.com/foo.html">');
    });

    test('ignoreLinks option', (done) => {
      opts.query.ignoreLinks = [
        'foo.html',
        '/bar',
        /node_modules/,
      ];
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<link rel="import" href="foo.html">' +
         '<link rel="import" href="foofoo.html">' +
         '<link rel="import" href="/bar/foo.html">' +
         '<link rel="import" href="../../node_modules/some-module/some-element.html">',
      );
    });

    test('ignoreLinksFromPartialMatches option', (done) => {
      opts.query.ignoreLinksFromPartialMatches = ['foo.html'];
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<link rel="import" href="foo.html">' +
        '<link rel="import" href="foofoo.html">');
    });

    test('ignorePathReWrite option', (done) => {
      opts.query.ignorePathReWrite = ['foo.html'];
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<link rel="import" href="foo.html">' +
        '<link rel="import" href="foofoo.html">');
    });

    test('remove leading ~', (done) => {
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<link rel="import" href="~foo/foo.html">');
    });
  });

  describe('domModule', () => {
    test('transforms dom-modules', (done) => {
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<dom-module id="x-foo">' +
        '<div></div></dom-module>');
    });

    test('transforms multiple dom-modules', (done) => {
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<dom-module id="x-foo">' +
        '<div></div></dom-module><dom-module id="x-foo-foo">' +
        '<div></div></dom-module>');
    });

    test('ignore non root level dom-modules', (done) => {
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<template><dom-module id="x-foo">' +
        '<div></div></dom-module></template>');
    });

    test('ignores invalid HTML', (done) => {
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '</td>');
    });

    test('ignore script tags in a template', (done) => {
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<dom-module id="x-foo"><template>' +
        '<script>var x = 1;</script></template></dom-module>');
    });

    test('removes script tags without a source', (done) => {
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).not.toBe(undefined);
        verifySourceMap(source, map);
        done();
      };
      loader.call(opts, '<dom-module id="x-foo">' +
        '<script>var x = 1;</script></dom-module>');
    });

    test('removes script tags without a protocol', (done) => {
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<dom-module id="x-foo">' +
        '<script src="foo.js"></script></dom-module>');
    });

    test('removes link tags', (done) => {
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<link rel="import" href="test.html">' +
        '<dom-module id="x-foo"></dom-module>');
    });

    test('keeps css link tags with import', (done) => {
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<dom-module id="x-foo">' +
        '<link rel="import" type="css" href="test.css"></dom-module>');
    });

    test('keeps css link tags with rel stylesheet', (done) => {
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<dom-module id="x-foo">' +
        '<link rel="stylesheet" href="test.css"></dom-module>');
    });

    test('adds to body if no dom-module', (done) => {
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<span></span>');
    });

    test('maintains links to stylesheet with an external url file', (done) => {
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<dom-module><template><link rel="stylesheet" href="http://example.com/test.css"></link></template></dom-module>');
    });

    test('maintains links to stylesheet with an protocol neutral href', (done) => {
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<dom-module><template><link rel="stylesheet" href="//example.com/test.css"></link></template></dom-module>');
    });

    test('ignores css link if flag is not set', (done) => {
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<dom-module><template><link rel="stylesheet" href="./test.css"></link></template></dom-module>');
    });

    test('ignores css link if flag is not set', (done) => {
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<dom-module><template><link rel="stylesheet" href="./test.css"></template></dom-module>');
    });

    test('rewrites css link tags with rel stylesheet', (done) => {
      opts.query.processStyleLinks = 'true';
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<dom-module id="x-foo"><template><link rel="stylesheet" href="test.css"></template></dom-module>');
    });

    test('rewrites css link tags with rel import', (done) => {
      opts.query.processStyleLinks = true;
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<dom-module id="x-foo"><template><link rel="import" type="css" href="test.css"></template></dom-module>');
    });

    test('rewrites multiple css link tags', (done) => {
      opts.query.processStyleLinks = true;
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<dom-module id="x-foo"><link rel="stylesheet" href="test1.css"><template><link rel="stylesheet" href="test2.css"></template></dom-module>');
    });
  });

  describe('scripts', () => {
    test('transforms scripts with a source into imports', (done) => {
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<script src="foo.js"></script>');
    });

    test('maintains external scripts', (done) => {
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<script src="http://example.com/test.js"></script>');
    });

    test('maintains inline scripts', (done) => {
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).not.toBe(undefined);
        verifySourceMap(source, map);
        done();
      };
      loader.call(opts, `<script>var x = 5;
        function foobar(arg) {
          var y = 6;
        
        }
      </script>`);
    });
  });

  describe('html-loader', () => {
    test('image sources are replaced with require calls', (done) => {
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<img src="foo.jpg" />');
    });

    test('html is minimized when option is set', (done) => {
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      opts.query.htmlLoader.minimize = true;
      loader.call(opts, '<script src="http://example.com/test.js"></script>');
    });
  });

  describe('styles', () => {
    test('in body have url() calls replaced with require statements', (done) => {
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<style>* {background-image: url("foo.jpg");}</style>');
    });

    test('in templates have url() calls replaced with require statements', (done) => {
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, '<template><style>* {background-image: url("foo.jpg");}</style></template>');
    });

    test('font url() are properly formatted', (done) => {
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).toBe(undefined);
        done();
      };
      loader.call(opts, `<style>
        @font-face {
          font-family: 'MyWebFont';
          src: url('webfont.eot'); /* IE9 Compat Modes */
          src: url('webfont.eot?#iefix') format('embedded-opentype'), /* IE6-IE8 */
              url('webfont.woff2') format('woff2'), /* Super Modern Browsers */
              url('webfont.woff') format('woff'), /* Pretty Modern Browsers */
              url('webfont.ttf')  format('truetype'), /* Safari, Android, iOS */
              url('webfont.svg#svgFontName') format('svg'); /* Legacy iOS */
        }
      </style>`);
    });
  });

  describe('full components', () => {
    test('external stylesheets', (done) => {
      opts.query.processStyleLinks = true;
      opts.async = () => (err, source, map) => {
        expect(err).toBe(null);
        expect(normalisePaths(source)).toMatchSnapshot();
        expect(map).not.toBe(undefined);
        verifySourceMap(source, map);
        done();
      };
      loader.call(opts, `<link rel="import" href="../bower_components/polymer/polymer-element.html">
      
      <dom-module id="my-element">
        <link rel="stylesheet" href="outside.css">
        <template>
          <link rel="stylesheet" href="inside.css">
          <h1>Hello, World! It's [[today]].</h1>
        </template>
        <script>
          // Heyyyy, we're pulling in a Node module!
          import format from 'date-fns/format';
          
         class MyElement extends Polymer.Element {
            static get is() { return 'my-element'; }
            static get properties() {
              return {
                today: {
                  type: String,
                  value: function() {
                    return format(new Date(), 'MM/DD/YYYY');
                  }
                }
              }
            }
          }
      
         window.customElements.define(MyElement.is, MyElement);
        </script>
      </dom-module>`);
    });
  });
});
