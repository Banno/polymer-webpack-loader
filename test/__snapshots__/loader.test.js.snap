// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader can process basic input 1`] = `
"
const RegisterHtmlTemplate = require('polymer-webpack-loader/register-html-template');

RegisterHtmlTemplate.toBody(\\"<div></div>\\");
"
`;

exports[`loader can process without options 1`] = `
"
const RegisterHtmlTemplate = require('polymer-webpack-loader/register-html-template');

RegisterHtmlTemplate.toBody(\\"<div></div>\\");
"
`;

exports[`loader domModule adds to body if no dom-module 1`] = `
"
const RegisterHtmlTemplate = require('polymer-webpack-loader/register-html-template');

RegisterHtmlTemplate.toBody(\\"<span></span>\\");
"
`;

exports[`loader domModule ignore non root level dom-modules 1`] = `
"
const RegisterHtmlTemplate = require('polymer-webpack-loader/register-html-template');

RegisterHtmlTemplate.toBody(\\"<template><dom-module id=\\\\\\"x-foo\\\\\\"><div></div></dom-module></template>\\");
"
`;

exports[`loader domModule ignore script tags in a template 1`] = `
"
const RegisterHtmlTemplate = require('polymer-webpack-loader/register-html-template');

RegisterHtmlTemplate.register(\\"<dom-module id=\\\\\\"x-foo\\\\\\"><template><script>var x = 1;</script></template></dom-module>\\");
"
`;

exports[`loader domModule ignores css link if flag is not set 1`] = `
"
const RegisterHtmlTemplate = require('polymer-webpack-loader/register-html-template');

RegisterHtmlTemplate.register(\\"<dom-module><template><link rel=\\\\\\"stylesheet\\\\\\" href=\\\\\\"./test.css\\\\\\"></template></dom-module>\\");
"
`;

exports[`loader domModule ignores css link if flag is not set 2`] = `
"
const RegisterHtmlTemplate = require('polymer-webpack-loader/register-html-template');

RegisterHtmlTemplate.register(\\"<dom-module><template><link rel=\\\\\\"stylesheet\\\\\\" href=\\\\\\"./test.css\\\\\\"></template></dom-module>\\");
"
`;

exports[`loader domModule ignores invalid HTML 1`] = `""`;

exports[`loader domModule keeps css link tags with import 1`] = `
"
const RegisterHtmlTemplate = require('polymer-webpack-loader/register-html-template');

RegisterHtmlTemplate.register(\\"<dom-module id=\\\\\\"x-foo\\\\\\"><link rel=\\\\\\"import\\\\\\" type=\\\\\\"css\\\\\\" href=\\\\\\"test.css\\\\\\"></dom-module>\\");
"
`;

exports[`loader domModule keeps css link tags with rel stylesheet 1`] = `
"
const RegisterHtmlTemplate = require('polymer-webpack-loader/register-html-template');

RegisterHtmlTemplate.register(\\"<dom-module id=\\\\\\"x-foo\\\\\\"><link rel=\\\\\\"stylesheet\\\\\\" href=\\\\\\"test.css\\\\\\"></dom-module>\\");
"
`;

exports[`loader domModule maintains links to stylesheet with an external url file 1`] = `
"
const RegisterHtmlTemplate = require('polymer-webpack-loader/register-html-template');

RegisterHtmlTemplate.register(\\"<dom-module><template><link rel=\\\\\\"stylesheet\\\\\\" href=\\\\\\"http://example.com/test.css\\\\\\"></template></dom-module>\\");
"
`;

exports[`loader domModule maintains links to stylesheet with an protocol neutral href 1`] = `
"
const RegisterHtmlTemplate = require('polymer-webpack-loader/register-html-template');

RegisterHtmlTemplate.register(\\"<dom-module><template><link rel=\\\\\\"stylesheet\\\\\\" href=\\\\\\"//example.com/test.css\\\\\\"></template></dom-module>\\");
"
`;

exports[`loader domModule removes link tags 1`] = `
"
require('./test.html');

const RegisterHtmlTemplate = require('polymer-webpack-loader/register-html-template');

RegisterHtmlTemplate.register(\\"<dom-module id=\\\\\\"x-foo\\\\\\"></dom-module>\\");
"
`;

exports[`loader domModule removes script tags without a protocol 1`] = `
"
const RegisterHtmlTemplate = require('polymer-webpack-loader/register-html-template');

RegisterHtmlTemplate.register(\\"<dom-module id=\\\\\\"x-foo\\\\\\"></dom-module>\\");

require('./foo.js');
"
`;

exports[`loader domModule removes script tags without a source 1`] = `
"
const RegisterHtmlTemplate = require('polymer-webpack-loader/register-html-template');

RegisterHtmlTemplate.register(\\"<dom-module id=\\\\\\"x-foo\\\\\\"></dom-module>\\");

var x = 1;
"
`;

exports[`loader domModule rewrites css link tags with rel import 1`] = `
"
const RegisterHtmlTemplate = require('polymer-webpack-loader/register-html-template');

RegisterHtmlTemplate.register(\\"<dom-module id=\\\\\\"x-foo\\\\\\"><template><style>\\" + require(\\"./test.css\\") + \\"</style></template></dom-module>\\");
"
`;

exports[`loader domModule rewrites css link tags with rel stylesheet 1`] = `
"
const RegisterHtmlTemplate = require('polymer-webpack-loader/register-html-template');

RegisterHtmlTemplate.register(\\"<dom-module id=\\\\\\"x-foo\\\\\\"><template><style>\\" + require(\\"./test.css\\") + \\"</style></template></dom-module>\\");
"
`;

exports[`loader domModule rewrites multiple css link tags 1`] = `
"
const RegisterHtmlTemplate = require('polymer-webpack-loader/register-html-template');

RegisterHtmlTemplate.register(\\"<dom-module id=\\\\\\"x-foo\\\\\\"><template><style>\\" + require(\\"./test1.css\\") + \\"</style><style>\\" + require(\\"./test2.css\\") + \\"</style></template></dom-module>\\");
"
`;

exports[`loader domModule transforms dom-modules 1`] = `
"
const RegisterHtmlTemplate = require('polymer-webpack-loader/register-html-template');

RegisterHtmlTemplate.register(\\"<dom-module id=\\\\\\"x-foo\\\\\\"><div></div></dom-module>\\");
"
`;

exports[`loader domModule transforms multiple dom-modules 1`] = `
"
const RegisterHtmlTemplate = require('polymer-webpack-loader/register-html-template');

RegisterHtmlTemplate.register(\\"<dom-module id=\\\\\\"x-foo\\\\\\"><div></div></dom-module>\\");

RegisterHtmlTemplate.register(\\"<dom-module id=\\\\\\"x-foo-foo\\\\\\"><div></div></dom-module>\\");
"
`;

exports[`loader full components external stylesheets 1`] = `
"
require('../bower_components/polymer/polymer-element.html');

const RegisterHtmlTemplate = require('polymer-webpack-loader/register-html-template');

RegisterHtmlTemplate.register(\\"<dom-module id=\\\\\\"my-element\\\\\\">\\\\n        \\\\n        <template><style>\\" + require(\\"./outside.css\\") + \\"</style>\\\\n          <style>\\" + require(\\"./inside.css\\") + \\"</style>\\\\n          <h1>Hello, World! It's [[today]].</h1>\\\\n        </template>\\\\n        \\\\n      </dom-module>\\");


          // Heyyyy, we're pulling in a Node module!
          import format from 'date-fns/format';
          
         class MyElement extends Polymer.Element {
            static get is() { return 'my-element'; }
            static get properties() {
              return {
                today: {
                  type: String,
                  value: function() {
                    return format(new Date(), 'MM/DD/YYYY');
                  }
                }
              }
            }
          }
      
         window.customElements.define(MyElement.is, MyElement);
        
"
`;

exports[`loader html-loader html is minimized when option is set 1`] = `
"
const RegisterHtmlTemplate = require('polymer-webpack-loader/register-html-template');

RegisterHtmlTemplate.toBody(\\"<script src=http://example.com/test.js></script>\\");
"
`;

exports[`loader html-loader image sources are replaced with require calls 1`] = `
"
const RegisterHtmlTemplate = require('polymer-webpack-loader/register-html-template');

RegisterHtmlTemplate.toBody(\\"<img src=\\\\\\"\\" + require(\\"./foo.jpg\\") + \\"\\\\\\">\\");
"
`;

exports[`loader links ignoreLinks option 1`] = `
"
require('./foofoo.html');
"
`;

exports[`loader links ignoreLinksFromPartialMatches option 1`] = `""`;

exports[`loader links ignorePathReWrite option 1`] = `
"
require('foo.html');

require('./foofoo.html');
"
`;

exports[`loader links ignores import links with external href 1`] = `
"
const RegisterHtmlTemplate = require('polymer-webpack-loader/register-html-template');

RegisterHtmlTemplate.toBody(\\"<link rel=\\\\\\"import\\\\\\" href=\\\\\\"https://example.com/foo.html\\\\\\">\\");
"
`;

exports[`loader links ignores links with invalid href 1`] = `""`;

exports[`loader links ignores stylesheet links with external href 1`] = `
"
const RegisterHtmlTemplate = require('polymer-webpack-loader/register-html-template');

RegisterHtmlTemplate.toBody(\\"<link rel=\\\\\\"stylesheet\\\\\\" href=\\\\\\"https://example.com/foo.html\\\\\\">\\");
"
`;

exports[`loader links remove leading ~ 1`] = `
"
require('foo/foo.html');
"
`;

exports[`loader links transforms links 1`] = `
"
require('./foo.html');
"
`;

exports[`loader scripts maintains external scripts 1`] = `
"
const RegisterHtmlTemplate = require('polymer-webpack-loader/register-html-template');

RegisterHtmlTemplate.toBody(\\"<script src=\\\\\\"http://example.com/test.js\\\\\\"></script>\\");
"
`;

exports[`loader scripts maintains inline scripts 1`] = `
"
var x = 5;
        function foobar(arg) {
          var y = 6;
        
        }
      
"
`;

exports[`loader scripts transforms scripts with a source into imports 1`] = `
"
require('./foo.js');
"
`;

exports[`loader styles font url() are properly formatted 1`] = `
"
const RegisterHtmlTemplate = require('polymer-webpack-loader/register-html-template');

RegisterHtmlTemplate.toBody(\\"<style>\\\\n        @font-face {\\\\n          font-family: 'MyWebFont';\\\\n          src: url('\\" + require(\\"./webfont.eot\\") + \\"'); /* IE9 Compat Modes */\\\\n          src: url('\\" + require(\\"./webfont.eot\\") + \\"?#iefix') format('embedded-opentype'), \\\\n              url('\\" + require(\\"./webfont.woff2\\") + \\"') format('woff2'), \\\\n              url('\\" + require(\\"./webfont.woff\\") + \\"') format('woff'), \\\\n              url('\\" + require(\\"./webfont.ttf\\") + \\"')  format('truetype'), \\\\n              url('\\" + require(\\"./webfont.svg\\") + \\"#svgFontName') format('svg'); /* Legacy iOS */\\\\n        }\\\\n      </style>\\");
"
`;

exports[`loader styles in body have url() calls replaced with require statements 1`] = `
"
const RegisterHtmlTemplate = require('polymer-webpack-loader/register-html-template');

RegisterHtmlTemplate.toBody(\\"<style>* {background-image: url('\\" + require(\\"./foo.jpg\\") + \\"');}</style>\\");
"
`;

exports[`loader styles in templates have url() calls replaced with require statements 1`] = `
"
const RegisterHtmlTemplate = require('polymer-webpack-loader/register-html-template');

RegisterHtmlTemplate.toBody(\\"<template><style>* {background-image: url('\\" + require(\\"./foo.jpg\\") + \\"');}</style></template>\\");
"
`;
